packages/api-client/src/index.server.ts

```diff
@@ -92,55 +95,88 @@ const tokenExtension: ApiClientExtension = {
   }
 };
 
+const injectClientTokens = async (context: ApiContext): Promise<ApiContext> => {
+  const [orderToken, bearerToken] = await Promise.all([
+    getCurrentCartToken(context.config),
+    getCurrentBearerToken(context)
+  ]);
+  return {
+    ...context,
+    client: context.client
+      .withBearerToken(bearerToken)
+      .withOrderToken(orderToken?.orderToken)
+  };
+};
+
+const withEndpointMiddleware = <RESPONSE = any>(
```

Comment:
This has been created in order to deal with async bearer token refreshment, which is not possible inside of `onCreate` or extension methods of the API client.